common/src/test/java/tachyon/security/authentication/AuthenticationFactoryTest.java

```diff
@@ -22,49 +42,235 @@
 
 import tachyon.Constants;
 import tachyon.conf.TachyonConf;
+import tachyon.util.network.NetworkAddressUtils;
 
 /**
  * Unit test for inner class {@link tachyon.security.authentication.AuthenticationFactory
  * .AuthType} and methods of {@link tachyon.security.authentication.AuthenticationFactory}
+ *
+ * In order to test methods that return kinds of TTransport for connection in different mode,
+ * we build Thrift servers and clients with specific TTransport, and let them connect.
  */
 public class AuthenticationFactoryTest {
 
+  TThreadPoolServer mServer;
+  TachyonConf mTachyonConf = new TachyonConf();
+  InetSocketAddress mServerAddress = new InetSocketAddress("localhost",
+      Constants.DEFAULT_MASTER_PORT);
+
   @Rule
   public ExpectedException mThrown = ExpectedException.none();
 
   @Test
   public void authenticationFactoryConstructorTest() {
-    TachyonConf tachyonConf = new TachyonConf();
-    tachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+    mTachyonConf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
 
-    AuthenticationFactory authenticationFactory = new AuthenticationFactory(tachyonConf);
+    AuthenticationFactory authenticationFactory = new AuthenticationFactory(mTachyonConf);
 
-    Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE.getAuthName(),
-        authenticationFactory.getAuthTypeStr());
+    Assert.assertEquals(AuthenticationFactory.AuthType.SIMPLE, authenticationFactory.getAuthType());
   }
 
   @Test
   public void getAuthTypeFromConfTest() {
-    TachyonConf tachyonConf = new TachyonConf();
     AuthenticationFactory.AuthType authType;
 
```

Comment:
Done. Since `authenticationFactoryConstructorTest` already cover this test, I merged these 2 cases together.